internal/cmd/commands/authenticate/authenticate.go

```diff
@@ -35,5 +48,42 @@ func (c *Command) Help() string {
 }
 
 func (c *Command) Run(args []string) int {
-	return cli.RunResultHelp
+	client, err := c.Client(base.WithNoTokenScope(), base.WithNoTokenValue())
+	if c.WrapperCleanupFunc != nil {
+		defer func() {
+			if err := c.WrapperCleanupFunc(); err != nil {
+				c.PrintCliError(fmt.Errorf("Error cleaning kms wrapper: %w", err))
+			}
+		}()
+	}
+	if err != nil {
+		c.PrintCliError(fmt.Errorf("Error creating API client: %w", err))
+		return base.CommandCliError
+	}
+
+	// Lookup the primary auth method ID in the global scope
+	aClient := authmethods.NewClient(client)
+	pri, err := getPrimaryAuthMethodId(c.Context, aClient, "global", "")
+	if err != nil {
+		c.PrintCliError(errors.New("Error looking up primary auth method ID for global scope. Try setting primary auth method for global scope, or use a auth method subcommand (see 'boundary authenticate -h' for available sub command usage)."))
+		return base.CommandUserError
+	}
+
+	c.FlagAuthMethodId = pri
+
+	switch {
+	case strings.HasPrefix(c.FlagAuthMethodId, "ampw"):
+		cmd := PasswordCommand{Command: c.Command}
+		cmd.Run([]string{})
+
+	case strings.HasPrefix(c.FlagAuthMethodId, "amoidc"):
+		cmd := OidcCommand{Command: c.Command}
+		cmd.Run([]string{})
+
+	default:
+		c.PrintCliError(fmt.Errorf("Invalid auth method ID prefix '%s'. Only 'ampw' (password) and 'amoidc' (OIDC) auth method prefixes are supported.", c.FlagAuthMethodId))
```

Comment:
This error is probably confusing to the user, as they didn't specify the auth method ID (we inferred it from the primary auth method, right?). When we add LDAP this could be happening to users on old CLIs, so maybe we should make this say something like "The primary auth method was of an unsupported type, please upgrade your Boundary version" or something. What do you think?

Reply:
👍

Reply:
+1. Also maybe say "...are `currently` supported"

Reply:
👍 Fixed